XMonad/Util/Hacks.hs

```diff
@@ -145,3 +158,49 @@ trayAbovePanelEventHook trayQ panelQ ConfigureEvent{ev_window = w, ev_above = a}
     mapM_ (io . lowerWindow dpy) panelWs
   mempty
 trayAbovePanelEventHook _ _ _ = mempty
+
+-- $resizeTrayer
+-- Communicate tray (trayer) resize events to xmobar so that space may
+-- be reserved on xmobar to house the tray.
+--
+-- Basic Usage, demonstrating Trayer:
+-- First, add to handleEventHook as follows:
+--
+-- > handleEventHook = handleEventHook def <+> Hacks.trayerResizeHook
+--
+-- Lastly, update xmobarrc as follows:
+--
+-- > Config { ...
+-- >        , commands = [ ...
+-- >                     , Run XPropertyLog "_XMONAD_TRAYPAD", ... ]
+-- >        , template = "%_XMONAD_LOG% }{ ... %_XMONAD_TRAYPAD%"
+-- >        }
+
+trayDefaultXpropLogName :: String
+trayDefaultXpropLogName = "_XMONAD_TRAYPAD"
+
+trayerResizeHook
+  :: (Event -> X All) -- ^ event hook
+trayerResizeHook = trayerResizeHook' trayDefaultXpropLogName
+
+trayerResizeHook'
+  :: String           -- ^ xmonadPropLog string to use
+  -> (Event -> X All) -- ^ event hook result
+trayerResizeHook' s = trayResizeHook (trayDefaultAction s) trayerQuery
+
+trayResizeHook
```

Comment:
For the sake of consistency, this should be `trayResizeHook'` since it's more general (so move the ')

Reply:
There is no trayResizeHook already, so why not use trayResizehook instead of trayResizeHook'?  Note the confusing prefix of tray vs trayer.  The prefix change is intended to indicate the move away from something trayer-specific.

Are you sure it should be changed?

Reply:
You're right, I overlooked it